storage/dataflux/worksteal.go

```diff
@@ -154,15 +153,15 @@ func (w *worker) doWorkstealListing(ctx context.Context) error {
 
 		// If listing not complete and idle workers are available, split the range
 		// and give half of work to idle worker.
-		if len(w.idleChannel)-len(w.lister.ranges) > 0 && ctx.Err() == nil {
+		for len(w.idleChannel)-len(w.lister.ranges) > 0 && ctx.Err() == nil {
```

Comment:
- The example looks perfect.  
-    The distribution of work affects the performance by large for real world data. There is a higher chance of next object names being closer to start of the range. And if its far, one of the passed split will get to it. It is much faster splitting work in 1/2, 1/4,1/8... manner.

- We don't want to split and add work to the queue if there is pending work in range channel and the idle workers are in sleep at the point. This will check number of workers that can pick the work and work that is already pending. Similar case would be when we call NextBatch for the second time, the range channel would be full due to remaining work in the previous run and all the workers may be idle too. This create more work when all the workers have not picked the work yet.

Reply:
I see, that makes a lot more sense now. Thanks Akansha!